src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -129,23 +134,27 @@ static ucs_status_t uct_cuda_copy_iface_flush(uct_iface_h tl_iface, unsigned fla
 }
 
 static UCS_F_ALWAYS_INLINE unsigned
-uct_cuda_copy_progress_event_queue(ucs_queue_head_t *event_queue, unsigned max_events)
+uct_cuda_copy_progress_event_queue(uct_cuda_copy_iface_t *iface,
+                                   uct_cuda_copy_stream_t index,
+                                   unsigned max_events)
 {
     unsigned count = 0;
     cudaError_t result = cudaSuccess;
     uct_cuda_copy_event_desc_t *cuda_event;
     ucs_queue_iter_t iter;
 
-    ucs_queue_for_each_safe(cuda_event, iter, event_queue, queue) {
+    ucs_queue_for_each_safe(cuda_event, iter,
+                            &iface->outstanding_event_q[index], queue) {
         result = cudaEventQuery(cuda_event->event);
         if (cudaSuccess != result) {
             break;
         }
-        ucs_queue_del_iter(event_queue, iter);
+        ucs_queue_del_iter(&iface->outstanding_event_q[index], iter);
         if (cuda_event->comp != NULL) {
             uct_invoke_completion(cuda_event->comp, UCS_OK);
         }
         ucs_trace_poll("CUDA Event Done :%p", cuda_event);
+        iface->stream_refcount[cuda_event->stream_id]--;
```

Comment:
isn't stream_id == index?
also should not use the name 'index' as it's name of gcc function, it breaks PGI compilation